plugins/inputs/ravendb/ravendb.go

```diff
@@ -0,0 +1,680 @@
+}
+
+type collectionMetrics struct {
+	DatabaseName          string `json:"DatabaseName"`
+	CollectionName        string `json:"CollectionName"`
+	DocumentsCount        int64  `json:"DocumentsCount"`
+	TotalSizeInBytes      int64  `json:"TotalSizeInBytes"`
+	DocumentsSizeInBytes  int64  `json:"DocumentsSizeInBytes"`
+	TombstonesSizeInBytes int64  `json:"TombstonesSizeInBytes"`
+	RevisionsSizeInBytes  int64  `json:"RevisionsSizeInBytes"`
+}
+
+var sampleConfig = `
+  ## Node URL and port that RavenDB is listening on.
+  url = "https://localhost:8080"
+
+  ## RavenDB X509 client certificate setup
+  tls_cert = "/etc/telegraf/raven.crt"
+  tls_key = "/etc/telegraf/raven.key"
+
+  ## Optional request timeouts
+  ##
+  ## ResponseHeaderTimeout, if non-zero, specifies the amount of time to wait
+  ## for a server's response headers after fully writing the request.
+  # header_timeout = "3s"
+  ##
+  ## client_timeout specifies a time limit for requests made by this client.
+  ## Includes connection time, any redirects, and reading the response body.
+  # client_timeout = "4s"
+
+  ## When true, collect server stats
+  # gather_server_stats = true
+
+  ## When true, collect per database stats
+  # gather_db_stats = true
+
+  ## When true, collect per index stats
+  # gather_index_stats = true
+
+  ## When true, collect per collection stats
+  # gather_collection_stats = true
+
+  ## List of db where database stats are collected
+  ## If empty, all db are concerned
+  # db_stats_dbs = []
+
+  ## List of db where index status are collected
+  ## If empty, all indexes from all db are concerned
+  # index_stats_dbs = []
+
+  ## List of db where collection status are collected
+  ## If empty, all collections from all db are concerned
+  # collection_stats_dbs = []
+`
+
+// SampleConfig ...
+func (r *RavenDB) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description ...
+func (r *RavenDB) Description() string {
+	return "Reads metrics from RavenDB servers via the Monitoring Endpoints"
+}
+
+// Gather ...
+func (r *RavenDB) Gather(acc telegraf.Accumulator) error {
+	err := r.ensureClient()
+	if nil != err {
+		if nil != r.Log {
+			r.Log.Errorf("Error with Client %s", err)
+		}
+		return err
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(gatherFunctions))
+	for _, f := range gatherFunctions {
+		go func(gf gatherFunc) {
+			defer wg.Done()
+			gf(r, acc)
+		}(f)
+	}
+	wg.Wait()
+
+	return nil
+}
+
+func (r *RavenDB) ensureClient() error {
+	if r.Client == nil {
+
+		tlsCfg, err := r.ClientConfig.TLSConfig()
+		if err != nil {
+			return err
+		}
+		tr := &http.Transport{
+			ResponseHeaderTimeout: r.ResponseHeaderTimeout.Duration,
+			TLSClientConfig:       tlsCfg,
+		}
+		r.Client = &http.Client{
+			Transport: tr,
+			Timeout:   r.ClientTimeout.Duration,
+		}
+	}
+	return nil
+}
+
+func (r *RavenDB) requestJSON(u string, target interface{}) error {
+	if r.URL == "" {
+		r.URL = DefaultURL
+	}
+	u = fmt.Sprintf("%s%s", r.URL, u)
```

Comment:
Move this to an `Init()` function that is only called once. Check the input-plugin-example or interface for the signature. The `Init()` function is called once after parsing and setting the config parameters in the structure. This means you do this evaluation only once instead of doing it in every iteration cycle.

Reply:
I think I don't follow this url maybe different over time if you deal with multiple nodes.

Reply:
Sorry I don't understand your comment.

Reply:
I see you fixed this already. :+1: Please mark as resolved.